src/routes/movie-list-page/movie-list-page.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { lazy, Suspense, useEffect, useState } from 'react';
+import MovieDetails from '../../components/movie-details/movie-details';
+import Dialog from '../../components/dialog/dialog';
+import FormMovie from '../../components/forms/form-movie/form-movie';
+import GenreSelect from '../../components/genre-select/genre-select';
+import SortControl from '../../components/sort-control/sort-control';
+import DeleteMovie from '../../components/forms/delete-movie/delete-movie';
+import MovieTile from '../../components/movie-tile/movie-tile';
+import './movie-list-page.scss';
+import {
+  getMovies,
+  searchMoviesByGenre,
+  searchMoviesByTitle,
+  sortMovies
+} from '../../services/services';
+import { useSearchParams, useLoaderData } from 'react-router-dom';
+import { DefaultHeader } from '../../components/default-header/default-header';
+
+const MovieContextEnum = {
+  edit: 'edit',
+  delete: 'delete'
+};
+const urlSearchParams = new URLSearchParams();
+
+const MovieListPage = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [searchQuery, setSearchQuery] = useState(searchParams.get('query') || '');
+  const [sortCriterion, setSortCriterion] = useState(searchParams.get('sortBy') || '');
+  const [activeGenre, setActiveGenre] = useState(searchParams.get('genre') || '');
+  const [movieList, setMovieList] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState(useLoaderData());
+  const [modalAddMovie, setModalAddMovie] = useState(false);
+  const [modalDeleteMovie, setModalDeleteMovie] = useState(false);
+  const [modalEditMovie, setModalEditMovie] = useState(false);
+
+  const sortBy = ['release date', 'title'];
+  const genreList = ['all', 'Documentary', 'Comedy', 'Horror', 'Crime'];
+  const contextMenu = Object.keys(MovieContextEnum);
+
+  // onInit
+  useEffect(() => {
+    const fetchData = async () => {
+      setMovieList(await getMovies());
+    };
+    lazy(fetchData());
+  }, []);
```

Comment:
it's more clear providing arguments in to the array so that the callback is invoked on params change - you can provide there searchQuery, sortCriteria and genries and take control over that in one place